tests/discover/filtering.sh

```diff
@@ -45,6 +45,170 @@ rlJournalStart
+#        rlRun "cd $path"
+#        ids_amount=$(tmt run -r discover --how fmf --fmf-id \
+#                     tests --name /tests/unit |
+#                     grep "name:" |
+#                     wc -l)
+#        tests_amount=1
+#        rlAssertEquals "Check that number of fmf-ids equals to tests number" \
+#                       "$ids_amount" "$tests_amount"
+#    rlPhaseEnd
+
+    rlPhaseStartTest "fmf-id (w/o url): check the test with --how=shell"
+        path="$(git rev-parse --show-toplevel)"
+        rlRun "cd $path/plans/sanity"
+        rlRun "tmt run -r test --name /lint/plans \
+                          plan --name /plans/sanity/lint \
+                          discover -h shell --fmf-id finish 2>&1 | \
+                          tee output" 2
+        rlAssertGrep "Error: no such option: --fmf-id" output -i
+    rlPhaseEnd
+
+    # Raise an exception if --fmf-id uses w/o --url and git root doesn't exist
+    rlPhaseStartTest "fmf-id (w/o url): and git root doesn't exist"
+        tmp_dir="$(mktemp -d)"
+        rlRun "cd $tmp_dir"
+        rlRun "tmt init --template base"
+        rlRun "tmt run -rdvvv discover -h fmf --fmf-id finish 2>&1 \
+               | tee output" 2
+        rlAssertGrep "\`tmt run discover --fmf-id\` without \`url\` option \
+can be used only within git repo." output
+        rlRun "rm -rf $tmp_dir"
+    rlPhaseEnd
+
+    # If in non-git directory exist plan w/ and w/o url, then only fmf-ids for
+    # the plan w/ url should be reported
+    rlPhaseStartTest "fmf-id (w/o and w/ url): non-git directory"
+        plan="plan -n /plans/example"
+        tmp_dir1="$(mktemp -d)"
+        tmp_dir2="$(mktemp -d)"
+        rlRun "cd $tmp_dir1"
+        rlRun "tmt init --template full"
+        rlRun "cd $tmp_dir2"
+        rlRun "tmt init --template base"
+        rlRun "cp plans/example.fmf $tmp_dir1/plans/non-url.fmf"
+        rlRun "cd $tmp_dir1"
+        # check "discover --fmf-id" shows the same tests as "tmt run discover"
+        rlRun "tmt run -r discover -h fmf --fmf-id finish | tee output"
+        rlRun "tmt run -v $plan discover | tee discover"
+        tests_list=$(tac discover |
+                     sed -n '/summary:/q;p')
+        url_discover=$(grep "url:" discover | awk '{print $2}')
+
+        for test in $tests_list; do
+            rlAssertGrep "$test" output
+        done
+
+        ids_amount=$(grep -o -i "name:" output | wc -l)
+        url_fmf_id=$(grep "url:" output | head -n 1 | awk '{print $2}')
+
+        tests_amount=$(echo $tests_list | wc -w)
+        rlAssertEquals "Check that number of fmf-ids equals to tests number" \
+                       "$ids_amount" "$tests_amount"
+        rlAssertEquals "Check url" "$url_discover" "$url_fmf_id"
+        rlRun "rm -rf $tmp_dir1 $tmp_dir2"
+    rlPhaseEnd
+
+    # Raise an exception if current dir doesn't have .fmf
+    rlPhaseStartTest "fmf-id (w/o url): current dir doesn't have fmf metadata"
```

Comment:
I experimented a bit with my proposal fmf-id /url changes and behaviour changed to raise  `--fmf-id` without `url` option can be used only within git repo.` instead which is imho more appropriate (one can specify plan using tmt options)

Reply:
This test covers only an exception when fmf.Tree can't be initialized. fmf.py#175